streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -161,7 +181,7 @@ public void process(final K key, final V1 value) {
                     //
                     // the condition below allows us to process the late record without the need
                     // to hold it in the temporary outer store
-                    if (timeTo < maxStreamTime) {
+                    if (internalOuterJoinFixDisabled || timeTo < maxStreamTime) {
```

Comment:
We can refactor it as:

```
if (!outerJoinWindowStore.isPresent() || timeTo < maxStreamTime) {
    context().forward(key, joiner.apply(key, value, null));
} else {

}
```

Then `internalOuterJoinFixDisabled` can just be a local variable instead of a class field.

Reply:
Done